### PR TITLE
Fix ignored implicit indent after `GROUP BY` clause

### LINKED ISSUE #4539: 2.0.0 - Implicit indent not working on group by
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff v2.0.0 gives an LT02 error when parsing a multi-line `group by` with a hanging/implicit indent with `allow_implicit_indents = True`.

### Expected Behaviour

I expected the SQL to parse without errors.

### Observed Behaviour

SQLFluff gave the following error message:

```
L:   7 | P:   9 | LT02 | Expected line break and indent of 4 spaces before 'a'.
                       | [layout.indent]
```

Related to https://github.com/sqlfluff/sqlfluff/issues/4467 ?

### How to reproduce

SQL file:
```
select
    a,
    b

from my_table

group by a,
    b
```

### Dialect

`ansi`

### Version

```
> sqlfluff --version
sqlfluff, version 2.0.0

> python --version
Python 3.9.9
```

### Configuration

```
[sqlfluff]
dialect = ansi

[sqlfluff:indentation]
allow_implicit_indents = True
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by borchero**
### Brief summary of the change made

In the spirit of #4583, this ought to resolve #4539.

### Are there any other side effects of this change that we should be aware of?

Different formatting when using `allow_implicit_indents = true`.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/fixtures/rules/std_rule_cases/LT02-indent.yml [hunk 1]
```diff
@@ -1815,6 +1815,84 @@ test_fix_untaken_positive_4433:
...
+  pass_str: |
+    SELECT t.col1, t.col2
+    FROM t
+    GROUP BY t.col1, t.col2
+  configs:
+    core:
+      dialect: ansi
+    indentation:
+      allow_implicit_indents: true
+
```
Comment (alanmcruickshank) [inline test/fixtures/rules/std_rule_cases/LT02-indent.yml:1829]: I'm a little confused about these test cases - they would still pass even _without_ implicit indents. 🤔 

To test the changes that you're making don't you need a line break before `t.col2` in the `GROUP BY` clause?

Comment (fredriv) [reply test/fixtures/rules/std_rule_cases/LT02-indent.yml:1829]: Was thinking the same thing 🙂

**Comment by fredriv** [comment]
The implicit indent wouldn't require you to take it, so with this change you'd be able to choose between

```
GROUP BY t.col1, t.col2

GROUP BY t.col1,
    t.col2

GROUP BY
    t.col1,
    t.col2
```
where the middle alternative is not allowed today. Right?

**Comment by WittierDinosaur** [comment]
I think as we roll more implicit indents out, we need a way to select in the config which implicit indents we enable. This one I think is the first controversial implementation. @alanmcruickshank what are your thoughts on this?

**Comment by alanmcruickshank** [comment]
> I think as we roll more implicit indents out, we need a way to select in the config which implicit indents we enable. This one I think is the first controversial implementation. @alanmcruickshank what are your thoughts on this?

I think you're right that this should be a configurable one. Happy to support it as an option, but I don't think it should be a default. The `Conditional` grammar should already allow this kind of thing.  What I'm not sure about is whether this is an isolated example (which would be configured with something like `implicit_group_by`) or whether this is an example of a wider pattern which might apply to other places like `ORDER BY`, `SELECT` or even more generally within delimited places (like lists or arrays) (which would be blanket enabled with something like `implicit_delimited`).

@fredriv @borchero - do either of you have views on whether this is an isolated example or something more general?

**Pull request closed without merging**